src/page/post_view/video_loader.rs

```diff
@@ -103,7 +110,56 @@ pub fn VideoView(idx: usize, muted: RwSignal<bool>) -> impl IntoView {
         Some(())
     });
 
+    #[cfg(feature = "hydrate")]
+    {
+        let (watched_percentage, set_watched_percentage) = create_signal(0_u8);
+        let (watched_count, set_watched_count) = create_signal(0_u8);
+
+        let _ = use_event_listener(container_ref, ev::timeupdate, move |event| {
+            let target = event.target().unwrap();
+            let video = target.unchecked_into::<web_sys::HtmlVideoElement>();
+            let duration = video.duration();
+            let current_time = video.current_time();
+
+            set_watched_percentage.update(|watched_percentage| {
+                *watched_percentage = (100.0 * (current_time / duration)) as u8;
+            });
+
+            if current_time == 0.0 {
```

Comment:
store the previous `current_time`, if current_time < previous_time

Reply:
done